lib/easypost/exceptions/api_error.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+require_relative 'easy_post_error'
+require 'easypost/constants'
+
+class EasyPost::Exceptions::ApiError < EasyPost::Exceptions::EasyPostError
+  attr_reader :status_code, :code, :errors
+
+  def initialize(message, status_code = nil, error_type = nil, errors = nil)
+    super message
+    @status_code = status_code
+    @code = error_type
+    @errors = errors
+  end
+
+  def pretty_print
+    error_string = "#{code} (#{status_code}): #{message}"
+    errors&.each do |error|
+      error_string += "\nField: #{error.field}\nMessage: #{error.message}"
+    end
+    error_string
+  end
+
+  def self.get_potential_error(response)
```

Comment:
issue: I don't feel this function name describes exactly what happens here, the whole `potential` bit is confusing. Could we follow the patterns of other libs and call it something like `handle_api_error` or something similar?